### PR TITLE
Added support for BatchToSpace and SpaceToBatch

### PR DESCRIPTION
**Pull request by nitishat9**
Implemented BatchToSpace and SpaceToBatch
operation

Signed-off-by: nitishat9 <nitisha.tomar@intel.com>

***VTS***
=============== Summary ===============
Total Run time: 10m 21s
2/2 modules completed
Total Tests       : 29912
PASSED            : 10970
FAILED            : 18
IGNORED           : 18924
============== End of Results ==============

***CTS***
=============== Summary ===============
Total Run time: 1h 21m 36s
2/2 modules completed
Total Tests       : 130076
PASSED            : 127226
FAILED            : 2850
============== End of Results ==============

### CODE REVIEW

File: ngraph_creator/operations/src/Batch_To_Space.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    if (layout) useNchw = true;
+
+    std::vector<uint32_t> shape(inDims.size(), 0);
+
+    auto block_shape_node =
+        createConstNode(ngraph::element::i32, ngraph::Shape{inDims.size()}, block_shape);
+    auto crop_begin = createConstNode(ngraph::element::i32, ngraph::Shape{shape.size()}, shape);
+    auto crop_end = createConstNode(ngraph::element::i32, ngraph::Shape{shape.size()}, shape);
+
+    if (mNgraphNodes->isForcedNchw(inputIndex)) {
```
Comment (Nagamani71) [inline ngraph_creator/operations/src/Batch_To_Space.cpp:80]: Remove this ForceNCHW check

File: ngraph_creator/operations/src/Space_To_Batch.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        createConstNode(ngraph::element::i32, ngraph::Shape{inDims.size()}, block_shape);
+    const auto crop_begin =
+        createConstNode(ngraph::element::i32, ngraph::Shape{inDims.size()}, crop_0);
+    const auto crop_end =
+        createConstNode(ngraph::element::i32, ngraph::Shape{inDims.size()}, crop_1);
+
+    if (inputsSize == 4) layout = sModelInfo->ParseOperationInput<uint8_t>(mNnapiOperationIndex, 3);
+    if (layout) useNchw = true;
+
+    if (mNgraphNodes->isForcedNchw(inputIndex)) {
```
Comment (Nagamani71) [inline ngraph_creator/operations/src/Space_To_Batch.cpp:110]: Remove isForcedNchw check

**Review by akodanka — CHANGES_REQUESTED** [review]

File: ngraph_creator/operations/src/Space_To_Batch.cpp [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    const auto& block_shape_OperandIndex = sModelInfo->getOperationInput(mNnapiOperationIndex, 1);
+    auto block_shape = sModelInfo->GetConstVecOperand<int32_t>(block_shape_OperandIndex);
+
+    // Compensation for the shape to be same as the size of data input shape
+    block_shape.insert(block_shape.begin(), 1);
+    block_shape.insert(block_shape.begin(), 1);
+
+    const auto& crop_OperandIndex = sModelInfo->getOperationInput(mNnapiOperationIndex, 2);
+    // Fetch the 2D crop as a 1D vector, and then split it into 2
+    auto crop_2d = sModelInfo->GetConstVecOperand<int32_t>(crop_OperandIndex);
```
Comment (akodanka) [inline ngraph_creator/operations/src/Space_To_Batch.cpp:80]: Why are we using the name crop here ? Aren't these paddings ?

Comment (nitishat9) [reply ngraph_creator/operations/src/Space_To_Batch.cpp:80]: Crop is the name being used in the openvino ngraph operation usage. So named it based on that 
Reference : https://docs.openvinotoolkit.org/latest/openvino_docs_ops_movement_BatchToSpace_2.html
Let me know if this needs to be changed

File: ngraph_creator/operations/src/Batch_To_Space.cpp [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    std::vector<uint32_t> shape(inDims.size(), 0);
+
+    const auto block_shape_node =
+        createConstNode(ngraph::element::i64, {inDims.size()}, block_shape);
+    const auto crop_begin = createConstNode(ngraph::element::i64, {shape.size()}, shape);
+    const auto crop_end = createConstNode(ngraph::element::i64, {shape.size()}, shape);
+
+    if (!useNchw) {  // No conversion needed if useNchw set
+        inputNode = transpose(NHWC_NCHW, inputNode);
+        mNgraphNodes->setForcedNchw(mDefaultOutputIndex, true);
```
[the remark this answers is not in the document]
Comment (nitishat9) [reply ngraph_creator/operations/src/Batch_To_Space.cpp:79]: Nagamani had commneted to remove isForcedNchw check ....so kept the else part of it as we have a layout parameter for the data representation. Let me know if this needs to be removed too.

[the remark this answers is not in the document]
Comment (akodanka) [reply ngraph_creator/operations/src/Batch_To_Space.cpp:79]: The usage of *ForcedNchw needs to be removed.

File: ngraph_creator/operations/src/Space_To_Batch.cpp (hunk 3 above)
Comment (akodanka) [reply ngraph_creator/operations/src/Space_To_Batch.cpp:80]: For space to batch, the names are paddings*:
https://docs.openvinotoolkit.org/latest/openvino_docs_ops_movement_SpaceToBatch_2.html

**Review by akodanka — APPROVED** [review]
LGTM

**Pull request merged**